common/schema/config.go

```diff
@@ -17,6 +17,27 @@ type Configuration struct {
 	NotMainTableOnly []ColumnKey `validate:"ninterfield=MainTableOnly"`
 	// Materialize lists columns that shall be materialized at ingest instead of computed at query time
 	Materialize []ColumnKey
+	// CustomDictionaries allows enrichment of flows with custom metadata
+	CustomDictionaries map[string]CustomDict `validate:"omitempty,dive"`
+}
+
+// CustomDict represents a single custom dictionary
+type CustomDict struct {
+	Keys       []CustomDictElement `validate:"dive"`
+	Attributes []CustomDictElement `validate:"dive"`
+	Source     string
+	Layout     string `default:"hashed"`
+	Dimensions []string
+}
+
+// CustomDictElement represents a single column of a custom dictionary
+type CustomDictElement struct {
+	Name      string
+	Type      string
+	Label     string
+	Match     string
+	MatchPost string
+	Default   string
```

Comment:
Add some validation. `validate:"required,alphanum"` for  `Name`, `validate:"required,oneof=String UInt8 UInt16 UInt32 UInt64 IPv4 IPv6"` for `Type` (if we need to match, we need to not allow anything wild here). `validate:"alphanum"` for `Label`, `Match` and `MatchPost`.

Moreover, from my understanding, `Label` and `Default` are for values and `Match` and `MatchPost` for keys. I would prefer to have two seperate types for that (`CustomDictKey` and `CustomDictValue`).

Reply:
Improved with 542f36cb3cb5c771d1e408eeb8c9360806f7b7a2